src/platform/unix/mod.rs

```diff
@@ -592,23 +598,13 @@ impl OsIpcOneShotServer {
     pub fn new() -> Result<(OsIpcOneShotServer, String),UnixError> {
         unsafe {
             let fd = libc::socket(libc::AF_UNIX, SOCK_SEQPACKET, 0);
-            let mut path: Vec<u8>;
-            loop {
-                let path_string = CString::new(&b"/tmp/rust-ipc-socket.XXXXXX"[..]).unwrap();
-                path = path_string.as_bytes_with_nul().iter().cloned().collect();
-                if *mktemp(path.as_mut_ptr() as *mut c_char) == 0 {
-                    return Err(UnixError::last())
-                }
-
-                let (sockaddr, len) = new_sockaddr_un(path.as_ptr() as *const c_char);
-                if libc::bind(fd, &sockaddr as *const _ as *const sockaddr, len as socklen_t) == 0 {
-                    break
-                }
+            let temp_dir =  Builder::new().tempdir().unwrap();
+            let socket_path = temp_dir.path().join("socket");
+            let path_string = socket_path.to_str().to_owned().unwrap();
```

Comment:
`to_owned()` in this place is useless. My suggestion was to add it *after* the `unwrap()`, to convert the resulting `&str` into a `String`. (Maybe I should have said `to_string()` instead to make it clearer... Not sure whether there is consensus nowadays which one is preferable.)

Though doing the `to_string()` conversion only while returning the result, as you did, is fine too -- i.e. you don't need the extra `to_owned()` or `to_string()` here at all.

Reply:
No, that's what you had before... Your last version was actually closer.

Sorry if I'm not making myself clear. I was thinking of something like

```rust
    let name = socket_path.to_str().unwrap();
    let name_ffi = CString::new(name).unwrap();
```

(And later return `name.to_string()`.)

Maybe you could even do the `CString` conversion directly in the `new_sockaddr_un()` call, since you do not need the value of `name_ffi` anywhere else...

Reply:
@antrik : thank you so much for keep patience with me.